packages/xchain-thorchain/src/client.ts

```diff
@@ -68,14 +68,12 @@ class Client implements ThorchainClient, XChainClient {
   }
 
   setNetwork = (network: Network): XChainClient => {
-    this.network = network
-    this.thorClient = new CosmosSDKClient({
-      server: this.getClientUrl(),
-      chainId: this.getChainId(),
-      prefix: this.getPrefix(),
-      derive_path: this.derive_path,
-    })
-    this.address = ''
+    if (this.network != network) {
+      this.network = network
+      this.clientUrl = this.getDefaultClientUrl()
```

Comment:
^ That would deprecate/override `clientUrl`, which may set using `setClientUrl` before.